### ISSUE TITLE
Bulit-in Public Library where Developers can publish their Projects for players to test.

### ISSUE THREAD

**Issue by AAAGroth** [godotengine/godot-proposals#7735]
### Describe the project you are working on

"I'm primarily a growth hacker, specializing in brand growth strategies. In my free time, I pursue game development as a hobby.

### Describe the problem or limitation you are having in your project

**1. Understanding engine capabilities:** 

I believe it's essential to clearly understand what is achievable with your chosen game engine from the beginning.

I believe one of the reasons why Godot doesn't receive as much attention as it deserves, and why more people aren't aware of its capabilities, is because the most significant marketing tool for any game engine is the games developed with it. 

So, let me elaborate...

Recently, I discovered a realistic first-person shooter game built using Godot, and I was genuinely impressed by what the engine was capable of. 

Unfortunately, the developer decided to switch engines after investing almost two years of work into the project.

This raised a question for me: 

Why would someone dedicate years to a game only to switch engines after so much effort? 

Later I discovered that the developer cited issues with the way nodes work as the reason for the switch and someone who was coding with him left the project. 

I truly understand that someone leaving a main role can be frustrating, during my years, I've had that happen plenty of times.

However, I believe it's fair to say that if more people were following the project, there would be better chances for this person to find new collaborators and build a community that watches the progress. This, without a doubt, would inspire the developer to keep going, and it would certainly make it easier for them to find new people to collaborate with.

**2. Finding Collaborations for Game Development:**

I firmly believe that collaborations are especially vital for indie projects. There's a quote I frequently hear from a friend: "No one creates anything truly great alone." While it's possible to develop an entire game solo, particularly for smaller projects, involving more people not only accelerates the process but also elevates the overall quality. This is because you can concentrate on aspects that align with your talents and interests.

So, if you're seeking a collaboration opportunity, it's crucial to initiate work on the project beforehand. Share your progress on platforms like Discord, create a YouTube video to showcase your work, and actively reach out to potential collaborators.

However, there's a challenge here—posting a video on YouTube might garner only a handful of views. Moreover, by the time you discover potential projects to collaborate on, it might be too late, as in the case of the FPS game developer who decided to change engines.

This leads me to wonder if there are other hidden gem games created in Godot that remain undiscovered, hidden away on someone's SSD.

In summary, collaborations can be a game-changer for indie game developers. Nevertheless, finding the right partners and gaining visibility for your work can be challenging, even on platforms like YouTube. However, these platforms do contain their fair share of undiscovered gems, like the FPS game I recently stumbled upon. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

**Solution:**

Implementing a built-in system for developers to showcase their games and projects, like a store where people could playtest and have feedback.

I believe that Godot could greatly benefit from a feature that allows developers to showcase their projects easily for playtesting, collaborating, and feedback gathering. 

Such a feature would demonstrate how much Godot values its developers. 

So making developers feel loved would not only increase their loyalty to the Godot engine but also inspire them to continue their projects and contribute to the engine's growth.

In my opinion, Godot has the potential to be much more in the game development industry, even surpassing engines like Unity and GameMaker. However, it's currently lacking the means to attract and retain developers effectively to grow exponentially.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

There would be a new tab called "Public Projects" where developers can upload their projects to the cloud and share their contact information (such as Discord and email). It would function like a store where other developers or players can playtest and see what the engine is capable of.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't have the expertise to answer.

### Is there a reason why this should be core and not an add-on in the asset library?

Even though they are similar in a way, they have different purposes. Therefore, they would need to be in a different place and not inside the asset library.

**Comment by Calinou** [comment]
This already exists in the form of the [asset library](godotengine.org/asset-library/), which has a Projects section :slightly_smiling_face: 

Anything else should be handled by community platforms, not as an official effort.

**Comment by rainlizard** [comment]
Calinou's link is broken. https://godotengine.org/asset-library/

I think the asset library could do with a lot of improvements. Especially these days with all the comparisons with Unity's asset store. The category system is very limiting, like what about a category for GDExtensions? I don't mean to sidetrack this proposal - but if for example the entire Category section was replaced with user tags (like Steam's tag system), then you could have any user-defined categories such as what this proposal is asking for. You could then have a `requesting_playtesting` tag.

**Comment by Calinou** [comment]
> The category system is very limiting, where's the category for GDExtensions? I don't mean to sidetrack this proposal - but if for example the entire Category section was replaced with user tags (like Steam's tag system), then you could have any user-defined categories such as what this proposal is asking for. You could then have a requesting_playtesting tag.

Unfortunately, doing this would require a significant rework of the current backend, which isn't in the cards for the reasons explained in https://github.com/godotengine/godot-asset-library/issues/302.

**Comment by jmarceno** [comment]
I don´t think this should happen inside the engine. But It is a good idea for a community site like GodotShaders.com.

My main reasons:
1. That are some forums on the internet where you can ask people to try your game. (Publishing on itch and asking on reddit always gives you plenty of very good feedback)
2. Having a mkt tool inside the engine, can incentive a lot of abuse and spam. (And it will be viewed as a mkt tool)
3. Number 2 can lead to a huge influx of players (not developers) that will see it as free game repo. Not bad per se, but will cause a lot maintenance overhead, with tons of "false" (sorry could not find a better word, I don´t want to imply malice here, just lack of knowledge on dev tools) issues being opened
4. IF number 3 happens as I speculate (and I really believe it will happen, as players these days love to see/test games in development, some even pay for that), those same players will/play those very unfinished games and correlate than with the engine itself.

**Comment by mellonedain** [comment]
Hi everybody,

I think, at the moment the biggest problem with the current library is impossible to cover multiple versions of assets and extensions dedicated for different Godot Engine versions. Please try to install Git extension in Godot 4.1.1...

I think, the Asset Library should be integrated with the Godot Engine itself. It is like https://packagist.org for PHP Composer, PyPi.org for Python or https://www.npmjs.com/package/package for NodeJS.

From programmer side it is more important to use libraries the same alone.

In my opinion we should improve the library code itself before we start rewrite it. We should remove the biggest problems and issues first and then start to work with something new / something bigger. 

The repository should be maintained by the Godot Foundation. As a developers we need to be sure, that any personal activity doesn't break it at all.

I am know PHP from the version 5. The code isn't bad because is simply and do what they need. Last two years I am working with versions 5.6, 7.0 and 8.*.  I am working with Zend Framework 1 and 2 and few authored frameworks. I can help If You need.

BTW: Mentioned GodotShaders.com is a Wordress site. With warnings at the moment...

**Comment by Calinou** [comment]
@mellonedain This is off-topic for this particular proposal. Also, we already plan to have an asset store as mentioned in the above comments.

**Comment by mellonedain** [comment]
Could you give me a link?
